crates/storage/provider/src/traits/executor.rs

```diff
@@ -33,13 +69,31 @@ pub trait BlockExecutor<SP: StateProvider> {
         block: &Block,
         total_difficulty: U256,
         senders: Option<Vec<Address>>,
-    ) -> Result<PostState, BlockExecutionError>;
+    ) -> Result<(), BlockExecutionError>;
 
-    /// Executes the block and checks receipts
+    /// Executes the block and checks receipts.
     fn execute_and_verify_receipt(
         &mut self,
         block: &Block,
         total_difficulty: U256,
         senders: Option<Vec<Address>>,
-    ) -> Result<PostState, BlockExecutionError>;
+    ) -> Result<(), BlockExecutionError>;
+
+    /// Return bundle state. This is output of the execution.
```

Comment:
should note that this is the output of execution for all executed blocks in between calling `take_output_state` i think